### PR TITLE
Add ampcombi 2.0.1

### PR DESCRIPTION
**Pull request by Darcy220606**
## PR checklist

This PR is dependent on ampcombi2 module update in https://github.com/nf-core/modules/pull/7142, once that is approved the version bump can be done in this PR as well. This was tested locally with manual version bump.

This PR also addresses and closes issue #423.

👽👽👽👽👽👽👽👽👽 Still TODO
- [x] Check merge_taxa script for dependency after updating ampcombi version in modules 
- Run the tests again to update the snapshots

------
- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Comment by jfy133** [comment]
Uhhh that's interesting... why don't we have any CI tests :sweat_smile:

File: bin/ampcombi_download.py [hunk 1]
```diff
@@ -1,78 +1,144 @@
...
+# Authors: [Anan Ibrahim](https://github.com/Darcy220606/AMPcombi), [Louisa Perelo](https://github.com/louperelo)
 # File: amp_database.py
 # Source: https://github.com/Darcy220606/AMPcombi/blob/main/ampcombi/amp_database.py
-# Source+commit: https://github.com/Darcy220606/AMPcombi/commit/a75bc00c32ecf873a133b18cf01f172ad9cf0d2d/ampcombi/amp_database.py
-# Download Date: 2023-03-08, commit: a75bc00c
 # This source code is licensed under the MIT license
 #########################################
 
-# TITLE: Download the DRAMP database if input db empty AND and make database compatible for diamond
+# TITLE: Download the reference database specified by the user.
```
Comment (jfy133) [inline bin/ampcombi_download.py:10]: Shouldn't this script be wrapped up within AMPcombi itself, e.g. `ampcombi download ...`?

File: docs/output.md [hunk 2]
```diff
@@ -443,9 +443,7 @@ Note that filtered FASTA is only used for BGC workflow for run-time optimisation
 
 [GECCO](https://gecco.embl.de) is a fast and scalable method for identifying putative novel Biosynthetic Gene Clusters (BGCs) in genomic and metagenomic data using Conditional Random Fields (CRFs).
 
-### Summary tools
-
-[AMPcombi](#ampcombi), [hAMRonization](#hamronization), [comBGC](#combgc), [MultiQC](#multiqc), [pipeline information](#pipeline-information), [argNorm](#argnorm).
+### Summary too](#ampcombi), [hAMRonization](#hamronization), [comBGC](#combgc), [MultiQC](#multiqc), [pipeline information](#pipeline-information), [argNorm](#argnorm).
```
Comment (jfy133) [inline docs/output.md:446]: The list of tools should not be in a header, this markdown looks a bit borked... can you doulbve check

File: docs/output.md [hunk 3]
```diff
@@ -510,7 +508,9 @@ Note that filtered FASTA is only used for BGC workflow for run-time optimisation
...
+[AMPcombi](https://github.com/Darcy220606/AMPcombi) summarizes the results of **antimicrobial peptide (AMP)** prediction tools (ampir, AMPlify, Macrel, and other non-nf-core tools) into a single table and aligns the hits against a reference AMP database for functional, structural and taxonomic classification using [MMseqs2](https://github.com/soedinglab/MMseqs2). It further assigns the physiochemical properties (e.g. hydrophobicity, molecular weight) using the [Biopython toolkit](https://github.com/biopython/biopython) and clusters the resulting AMP hits from all samples using [MMseqs2](https://github.com/soedinglab/MMseqs2). To further filter the recovered AMPs using the presence of signaling peptides, the output file `Ampcombi_summary_cluster.tsv` or `ampcombi_complete_summary_taxonomy.tsv.gz` can be used downstream as detailed [here](https://ampcombi.readthedocs.io/en/main/usage.html#signal-peptide). The final tables generated may also be visualized and explored using an interactive [user interface](https://ampcombi.readthedocs.io/en/main/visualization.html).
```
Comment (jfy133) [inline docs/output.md:511]:
```suggestion
[AMPcombi](https://github.com/Darcy220606/AMPcombi) summarizes the results of **antimicrobial peptide (AMP)** prediction tools (ampir, AMPlify, Macrel, and other non-nf-core supported tools) into a single table and aligns the hits against a reference AMP database for functional, structural and taxonomic classification using [MMseqs2](https://github.com/soedinglab/MMseqs2). It further assigns the physiochemical properties (e.g. hydrophobicity, molecular weight) using the [Biopython toolkit](https://github.com/biopython/biopython) and clusters the resulting AMP hits from all samples using [MMseqs2](https://github.com/soedinglab/MMseqs2). To further filter the recovered AMPs using the presence of signaling peptides, the output file `Ampcombi_summary_cluster.tsv` or `ampcombi_complete_summary_taxonomy.tsv.gz` can be used downstream as detailed [here](https://ampcombi.readthedocs.io/en/main/usage.html#signal-peptide). The final tables generated may also be visualized and explored using an interactive [user interface](https://ampcombi.readthedocs.io/en/main/visualization.html).
```

File: docs/usage.md [hunk 4]
```diff
@@ -224,14 +224,33 @@ wget https://github.com/nf-core/funcscan/raw/<PIPELINE_VERSION>/bin/ampcombi_dow
 python3 ampcombi_download.py
 ```
 
-However, the user can also supply their own custom AMP database by following the guidelines in [AMPcombi](https://github.com/Darcy220606/AMPcombi).
+IN addition to [DRAMP](http://dramp.cpu-bioinfor.org/), two more reference databases can be used to classify the recovered AMPs in the AMP workflow; [APD](https://aps.unmc.edu/) and [UniRef100](https://academic.oup.com/bioinformatics/article/23/10/1282/197795). Only one database can be used at a time using `--amp_ampcombi_db database_name`.
```
Comment (jfy133) [inline docs/usage.md:227]:
```suggestion
In addition to [DRAMP](http://dramp.cpu-bioinfor.org/), two more reference databases can be used to classify the recovered AMPs in the AMP workflow; [APD](https://aps.unmc.edu/) and [UniRef100](https://academic.oup.com/bioinformatics/article/23/10/1282/197795). Only one database can be used at a time using `--amp_ampcombi_db database_name`.
```

File: docs/usage.md [hunk 5]
```diff
@@ -224,14 +224,33 @@ wget https://github.com/nf-core/funcscan/raw/<PIPELINE_VERSION>/bin/ampcombi_dow
...
+    ├── ref_DB.dbtype
+    ├── ref_DB_h
+    ├── ref_DB_h.dbtype
+    ├── ref_DB_h.index
+    ├── ref_DB.index
+    ├── ref_DB.lookup
+    └── ref_DB.source
+```
+
+🗒️ **Note**: For both [DRAMP](http://dramp.cpu-bioinfor.org/) and [APD](https://aps.unmc.edu/), AMPcombi removes entries that contains any non amino acid residues by default.
```
Comment (jfy133) [inline docs/usage.md:253]:
```suggestion
:::note
For both [DRAMP](http://dramp.cpu-bioinfor.org/) and [APD](https://aps.unmc.edu/), AMPcombi removes entries that contains any non amino acid residues by default.
:::
```

You can use Matthias' website admonitions syntax: https://nf-co.re/docs/contributing/website/markdown#admonitions

File: nextflow.config [hunk 6]
```diff
@@ -119,13 +119,14 @@ params {
     amp_hmmsearch_savetargets                        = false
     amp_hmmsearch_savedomains                        = false
 
-    amp_ampcombi_db                                  = null
+    amp_ampcombi_db                                  = 'DRAMP'
```
Comment (jfy133) [inline nextflow.config:122]: maybe thiis shoul dbe `amp_ampcombi_db_id` to match with e.g. mmseqs parameters,  and just `amp_ampcombi_db` for the path, for consistency with the the other parameters in the pipeline

**Review by jfy133 — CHANGES_REQUESTED** [review]
AMPcombi itself is not updated in this PR ... is that correct?

**Comment by Darcy220606** [comment]
> Uhhh that's interesting... why don't we have any CI tests 😅

Was wondering the same, maybe because Im pushing into your branch and not the dev ?

File: bin/ampcombi_download.py (hunk 1 above)
Comment (Darcy220606) [reply bin/ampcombi_download.py:10]: It is but u remember the issue with it downloading the ref. database every time for each sample? We did that to only let it run once in the pipeline

**Comment by jfy133** [comment]
That shouldn't make s difference... But let's see once it's merged into the template update branch...

Ping me for re-review once you've updated and we see what happens

File: docs/output.md (hunk 2 above)
Comment (Darcy220606) [reply docs/output.md:446]: Fixed

File: docs/usage.md (hunk 5 above)
Comment (Darcy220606) [reply docs/usage.md:253]: ohh there is a whale too ❤️

File: nextflow.config (hunk 6 above)
Comment (Darcy220606) [reply nextflow.config:122]: Ok by me!

**Comment by Darcy220606** [comment]
@nf-core-bot fix linting

File: bin/ampcombi_download.py (hunk 1 above)
Comment (Darcy220606) [reply bin/ampcombi_download.py:10]: Also as i wrote in the PR comment, for this PR to run as it should we still need to update the module to v2.0.1, which has a sep. PR open in modules ;)

**Review by jfy133 — COMMENTED** [review]
@Darcy220606 where are the ampcombi 2.0 modules themselves? Or is this just the download script

**Comment by amizeranschi** [comment]
@jfy133 the Ampcombi 2.0 modules are in a separate PR, here: https://github.com/nf-core/modules/pull/7142

**Comment by jfy133** [comment]
Ah ok! Then we can wait for them to go in and then update this PR :)

**Comment by jfy133** [comment]
Initial review of that

**Comment by jfy133** [comment]
Still only thre checks?! Wtf?!

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
Let's assume it's OK...

**Comment by jfy133** [comment]
You can merge away @Darcy220606 !

**Pull request merged**